app/src/Upload.elm

```diff
@@ -34,16 +34,16 @@ main =
 
 
 type Model
-    = Waiting String
+    = Waiting String Int
```

Comment:
It would be much more flexible to use a record as a model for the data here, instead of requiring two arguments to Waiting.

Reply:
The Int should be a "Maybe Int", and it should be an own type alias called "NumReceivedGreetings" or similar.